code/modules/mob/living/carbon/human/species/outsider/undead.dm

```diff
@@ -406,13 +427,185 @@
 	grab_mod = 5
 
 	natural_armor = list(
-		ballistic = ARMOR_BALLISTIC_MEDIUM,
-		laser = ARMOR_LASER_PISTOL,
-		melee = ARMOR_MELEE_MAJOR,
+		ballistic = ARMOR_BALLISTIC_MINOR,
+		laser = ARMOR_LASER_MINOR,
+		melee = ARMOR_MELEE_SMALL,
 		bomb = ARMOR_BOMB_PADDED,
 		energy = ARMOR_ENERGY_SMALL
 	)
 
-	maneuvers = list(/singleton/maneuver/leap/areagrab)
+	inherent_verbs = list(
+						/mob/living/carbon/human/proc/trample,
+						/mob/living/carbon/human/proc/darkness_eyes,
+						/mob/living/carbon/proc/consume
+						)
 
 	unarmed_types = list(/datum/unarmed_attack/bite/infectious, /datum/unarmed_attack/golem)
+	show_ssd = TRUE
+
+/datum/species/zombie/rhino/get_random_name(gender)
+	return "zombie rhino"
+
+/mob/living/carbon/human/rhino/Initialize(mapload, new_species)
+	. = ..(mapload, SPECIES_ZOMBIE_RHINO)
+
+/mob/living/carbon/proc/consume()
+	set name = "Consume"
+	set desc = "Regain life and infect others by feeding upon them."
+	set category = "Abilities"
+
+	if (last_special > world.time)
+		to_chat(src, SPAN_WARNING("You aren't ready to do that! Wait [round(last_special - world.time) / 10] seconds."))
+		return
+
+	var/mob/living/carbon/human/target
+	var/list/victims = list()
+	for (var/mob/living/carbon/human/L in get_turf(src))
+		if (L != src && (L.lying || L.stat == DEAD))
+			if (iszombie(L))
+				to_chat(src, SPAN_WARNING("\The [L] isn't fresh anymore!"))
+				continue
+			if (!(L.species.zombie_type in list(SPECIES_ZOMBIE, SPECIES_ZOMBIE_BULL, SPECIES_ZOMBIE_HUNTER, SPECIES_ZOMBIE_RHINO, SPECIES_ZOMBIE_SKRELL, SPECIES_ZOMBIE_TAJARA, SPECIES_ZOMBIE_UNATHI)) || L.is_diona() || L.isSynthetic())
+				to_chat(src, SPAN_WARNING("You'd break your teeth on \the [L]!"))
+				continue
+			victims += L
+
+	if (!length(victims))
+		to_chat(src, SPAN_WARNING("No valid targets nearby!"))
+		return
+	if (client)
+		if (length(victims) == 1) //No need to choose
+			target = victims[1]
+		else
+			target = tgui_input_list(src, "Who would you like to consume?", "Zombie", victims)
+	else //NPCs
+		if (length(victims) > 0)
+			target = victims[1]
+
+	if (!target)
+		to_chat(src, SPAN_WARNING("You aren't on top of a victim!"))
+		return
```

Comment:
Seems redundant with what is below? Or at least, the chat message doesn't reflect the failure reason correctly?